tracer/src/Datadog.Trace/Iast/Aspects/System.Text/StringBuilderAspects.cs

```diff
@@ -1224,96 +1224,49 @@ public static StringBuilder AppendJoin(StringBuilder? target, char separator, ob
         return result;
     }
 
-    // TODO : Add support for callsites with undefined generic params
-/*
-
     /// <summary>StringBuilder.AppendJoin aspect</summary>
+    /// <typeparam name="T"> Joined element type </typeparam>
     /// <param name="target">The StringBuilder instance.</param>
     /// <param name="separator">The character to use as a separator.</param>
     /// <param name="values">An array that contains the strings to concatenate and append to the current instance of the string builder.</param>
     /// <returns>The modified StringBuilder instance.</returns>
-    [AspectMethodReplace("System.Text.StringBuilder::AppendJoin(System.Char,System.Collections.Generic.IEnumerable`1<!!0>)")]
-    public static StringBuilder AppendJoin(StringBuilder? target, char separator, object? values)
+    [AspectMethodReplaceFromVersion("3.2.0", "System.Text.StringBuilder::AppendJoin(System.Char,System.Collections.Generic.IEnumerable`1<!!0>)")]
+    public static StringBuilder AppendJoin<T>(StringBuilder? target, char separator, IEnumerable<T>? values)
     {
-        if (values is null || target is null)
-        {
-#pragma warning disable CS8604 // We want to call the exact method overload even if values is null
-            return target!.AppendJoin(separator, values as IEnumerable<object>);
-#pragma warning restore CS8604 // Enable
-        }
-
-        var valuesConverted = values as IEnumerable<object?>;
-
-        if (values is IEnumerable valuesEnumerable)
+        var result = target!.AppendJoin(separator, values!);
```

Comment:
That was the case before, but I'm confused why `StringBuilder` is marked as nullable in the signature, but then you immediately use `!`

Reply:
If the string builder is null the original call is going to fail (as intended) and that code will never execute